Tests/OTLPGRPCTests/Metrics/OTLPGRPCMetricExporterTests.swift

```diff
@@ -0,0 +1,147 @@
+                                    exemplars: [],
+                                    flags: []
+                                ),
+                            ]))
+                        ),
+                    ]
+                ),
+            ]
+        )
+
+        try await collector.withServer { endpoint in
+            let configuration = try OTLPGRPCMetricExporterConfiguration(
+                environment: [:],
+                endpoint: endpoint,
+                headers: [
+                    "key1": "42",
+                    "key2": "84",
+                ]
+            )
+            let exporter = OTLPGRPCMetricExporter(configuration: configuration)
+
+            try await exporter.export([resourceMetricsToExport])
+
+            await exporter.shutdown()
+        }
+
+        XCTAssertEqual(collector.metricsProvider.requests.count, 1)
+        let request = try XCTUnwrap(collector.metricsProvider.requests.first)
+
+        XCTAssertEqual(request.exportRequest.resourceMetrics.count, 1)
+        let resourceMetrics = try XCTUnwrap(request.exportRequest.resourceMetrics.first)
+        XCTAssertEqual(resourceMetrics.resource, .with {
+            $0.attributes = .init(["service.name": "mock_service"])
+        })
+        XCTAssertEqual(resourceMetrics.scopeMetrics.count, 1)
+        let scopeMetrics = try XCTUnwrap(resourceMetrics.scopeMetrics.first)
+        XCTAssertEqual(scopeMetrics.scope, .with {
+            $0.name = "scope_name"
+            $0.version = "scope_version"
+            $0.attributes = [
+                .with {
+                    $0.key = "scope_attr_key"
+                    $0.value = .init("scope_attr_val")
+                },
+            ]
+        })
+        XCTAssertEqual(scopeMetrics.metrics, .init(resourceMetricsToExport.scopeMetrics.first!.metrics))
+
+        XCTAssertEqual(request.headers.first(name: "key1"), "42")
+        XCTAssertEqual(request.headers.first(name: "key2"), "84")
+    }
+
+    func test_export_whenAlreadyShutdown_throwsAlreadyShutdownError() async throws {
+        let collector = OTLPGRPCMockCollector()
+        let errorCaught = expectation(description: "Caught expected error")
+        do {
+            try await collector.withServer { endpoint in
+                let configuration = try OTLPGRPCMetricExporterConfiguration(environment: [:], endpoint: endpoint)
+                let exporter = OTLPGRPCMetricExporter(configuration: configuration)
+                await exporter.shutdown()
+
+                let metrics = OTelResourceMetrics(scopeMetrics: [])
+                try await exporter.export([metrics])
+
+                XCTFail("Expected exporter to throw error, successfully exported instead.")
+            }
+        } catch is OTelMetricExporterAlreadyShutDownError {
+            errorCaught.fulfill()
+        }
+        await fulfillment(of: [errorCaught], timeout: 0.0)
+    }
+
+    func test_forceFlush() async throws {
```

Comment:
I'd rather tweak the Codecov config a bit further than add a non-testing test like this to increase code coverage. Does the PR fail if without this one already?

Reply:
I think it did, yes 🙃

Reply:
That's on me then, and I'll further tweak the configuration. This should definitely not be necessary, sorry that you keep on having to deal with it!